test/framework/resources.go

```diff
@@ -52,22 +51,23 @@ func allResourcesOfType(kumactl KumactlOptions, descriptor core_model.ResourceTy
 
 func deleteMeshResourcesKubernetes(cluster Cluster, mesh string, resource core_model.ResourceTypeDescriptor) error {
 	args := []string{"delete", strings.ReplaceAll(strings.ToLower(resource.PluralDisplayName), " ", "")}
-	if resource.IsPluginOriginated {
-		// because all new policies have a mesh label, we can just delete by selecting a label
-		args = append(args, "--all-namespaces", "--selector", fmt.Sprintf("%s=%s", mesh_proto.MeshTag, mesh))
-		if err := k8s.RunKubectlE(cluster.GetTesting(), cluster.GetKubectlOptions(), args...); err != nil {
-			return err
-		}
-	} else {
-		list, err := allResourcesOfType(*cluster.GetKumactlOptions(), resource, mesh)
-		if err != nil {
-			return err
-		}
-		for _, item := range list.GetItems() {
-			itemDelArgs := append(args, item.GetMeta().GetName())
-			if err := k8s.RunKubectlE(cluster.GetTesting(), cluster.GetKubectlOptions(), itemDelArgs...); err != nil {
+	list, err := allResourcesOfType(*cluster.GetKumactlOptions(), resource, mesh)
+	if err != nil {
+		return err
+	}
+	for _, item := range list.GetItems() {
+		var itemDelArgs []string
+		if resource.IsPluginOriginated {
+			name, ns, err := util_k8s.CoreNameToK8sName(item.GetMeta().GetName())
```

Comment:
This is a sad hack! Shouldn't just be a method on resource or something like this?

Reply:
Hmm, it's a k8s-specific method, I'm not sure if we should bring it into the resource. Kuma core model is not aware if the resource is `Cluster` or `Namespace` scoped.

Do you have anything specific in mind?

Reply:
Yeah I don't really have a solution. It just feels a little odd.